### PR TITLE
chore(build): fix gradle and detekt warnings

### PR DESCRIPTION
**Pull request by bas-info-nl**
Gradle fixes to make it future proof by removing use of deprecated var `rootDir`.
Applied recommended Detekt fixes for some typescript imports.

Resolves: PZ-4471

### CODE REVIEW

**Review by edgarvonk — APPROVED** [review]
Nice! But. where is the JIRA issue? ;-)

File: build.gradle.kts [hunk 1]
```diff
@@ -696,19 +703,20 @@ tasks {
...
         execGoal("wildfly-jar:package")
 
-        inputs.files(fileTree("$rootDir/src/main/resources/wildfly"))
-        inputs.file("$rootDir/build/libs/zaakafhandelcomponent.war")
-        inputs.file("$rootDir/pom.xml")
-        inputs.file("$rootDir/src/main/resources/wildfly/configure-wildfly.cli")
-        inputs.file("$rootDir/src/main/resources/wildfly/deploy-zaakafhandelcomponent.cli")
-        outputs.dir("$rootDir/target")
+        val wildflyResources = srcMainResources.dir("wildfly")
+        inputs.files(wildflyResources.asFileTree)
```
[the remark this answers is not in the document]
Comment (bas-info-nl) [reply build.gradle.kts:714]: O that's an interesting comment... I will see if there are any documented issues with that 🕵️

[the remark this answers is not in the document]
Comment (bas-info-nl) [reply build.gradle.kts:715]: When using `FileTree` as inputs or outputs in Gradle tasks, caching behavior is influenced by how Gradle tracks changes to the files within the tree. Here’s a breakdown of how it works:

### FileTree and Caching

1. **FileTree as Input**:
   - When you declare a `FileTree` as an input to a task, Gradle will monitor the contents of that tree. This means it checks for changes in the files (like modifications, additions, or deletions) to determine if the task needs to be executed again.
   - If the contents of the `FileTree` have not changed since the last build, Gradle can skip executing the task, leveraging the cached output from the previous execution.

2. **Performance Considerations**:
   - Using `FileTree` can have performance implications, especially if the tree is filtered or if it contains a large number of files. Gradle may need to traverse the directory structure multiple times to evaluate the inputs, which can slow down the build process.
   - It’s recommended to use `FileCollection` or `RegularFile` types when possible for better performance, as these are more efficient for caching checks.

3. **Output Caching**:
   - If a task produces outputs based on a `FileTree`, those outputs can also be cached. Gradle computes a cache key based on the task's inputs (including the `FileTree`) and its outputs. If the inputs haven’t changed, the cached outputs can be reused.
   - This means that if you run the task again without modifying the files in the `FileTree`, Gradle will use the cached output instead of re-executing the task.

### Best Practices

- **Use Specific File Types**: If you can, prefer using `RegularFile` or `Directory` types for inputs and outputs to improve caching efficiency.
- **Avoid Non-Deterministic Tasks**: Ensure that tasks using `FileTree` produce consistent outputs for the same inputs to benefit from caching effectively.

### Conclusion

In summary, Gradle handles `FileTree` inputs and outputs by monitoring changes to the files within the tree. If no changes are detected, it can skip task execution and reuse cached outputs, enhancing build performance. However, be mindful of performance impacts when using filtered file trees.

[the remark this answers is not in the document]
Comment (bas-info-nl) [reply build.gradle.kts:715]: With this knowledge, let me check if I can rewrite those inputs

[the remark this answers is not in the document]
Comment (bas-info-nl) [reply build.gradle.kts:715]: No issue here, the path only contains 2 files, the FileTree will be ok to handle that 👍
